### ISSUE TITLE
Fix font size and contrast of text across all of the admin

### ISSUE THREAD

**Issue by thibaudcolas** [wagtail/wagtail#5270]
### Issue Summary

Follow-up of #2224. Related: #5269. There are still a lot of places in the admin UI where text does not have enough contrast compared to its background, either because the text is too small (font size and weight), or the color does not have a good contrast ratio against the background color.

There is no WCAG 2.1 requirement for a specific font size, the requirement is for a [minimum contrast ratio](https://www.w3.org/TR/UNDERSTANDING-WCAG20/visual-audio-contrast-contrast.html).

Suggested approach:

- [x] Identify parts of the UI where fonts are too small / do not use the right color to provide enough contrast
- [x] Suggest increasing the size, weight, or change color
- [x] Potentially define a minimum font size to use across the whole UI

---

Here are the parts of the UI I could think of as being too small:

* [ ] Table headings
* [ ] Page status ("Live" / "Draft")
* [ ] “Modified” timestamp in page edit footer

For the contrast, here are parts of the UI that have been flagged in automated testing as too low contrast:

- [ ] "Release notes" link in the "Upgrade notification" block on the dashboard (admin only), path: /
- [ ] "Success" banner messages in the admin, white text and buttons on green background, e.g. after "Publish" on a page
- [ ] "Configure a site now" CTA at root level of the listing pages, path: /pages/
- [ ] "Warning" banner messages in the admin, white text and buttons on yellow background, e.g. when reviewing a revision /pages/60/revisions/37/revert/
- [ ] "Replace current draft" CTA of review revisions, path: /pages/60/revisions/37/revert/
- [ ] Pages "move" UI when some pages cannot receive sub-pages, 
- [ ] Page "parent" link in search results view of the link chooser, on hover
- [ ] "Max dimensions" and "File size" labels in image editing UI

There might be more that I missed. What the automated tools don't find is cases where the color contrast is good if you just compare text color with that of its background, but the text is too small or thin.

Annotated screenshots:

![Dashboard – Dashboard](https://user-images.githubusercontent.com/877585/57018604-13e26900-6c1c-11e9-8f8a-61ac969896b3.png)

![User account – Current time zone - Success](https://user-images.githubusercontent.com/877585/57018612-19d84a00-6c1c-11e9-9dfd-c93cb21eac54.png)

![Pages – View child pages - Root level](https://user-images.githubusercontent.com/877585/57018625-2361b200-6c1c-11e9-9954-ae74e4ced049.png)

![Pages – Review revision](https://user-images.githubusercontent.com/877585/57018637-2e1c4700-6c1c-11e9-8d7c-16556c8a78b7.png)

<img width="695" alt="review revisions cta" src="https://user-images.githubusercontent.com/877585/57018643-34aabe80-6c1c-11e9-923c-d3ca8a3272c7.png">

![Pages – Move](https://user-images.githubusercontent.com/877585/57018652-3a080900-6c1c-11e9-9529-a6ed8f40b097.png)

<img width="697" alt="Link chooser search page parent hover" src="https://user-images.githubusercontent.com/877585/57018671-4724f800-6c1c-11e9-83e3-9f3da644f591.png">

![Images – Edit](https://user-images.githubusercontent.com/877585/57018675-4be9ac00-6c1c-11e9-8eb6-d31f2df1cf67.png)
